src/test/java/com/example/java/maven/ticketVendingMachine/CoinStorageTest.java

```diff
@@ -0,0 +1,86 @@
+package com.example.java.maven.ticketVendingMachine;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.mockito.BDDMockito.then;
+
+import java.util.Arrays;
+import java.util.HashMap;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.times;
+
+
+public class CoinStorageTest {
+    MessagePrinter messagePrinter = Mockito.mock(MessagePrinter.class);
+
+    CoinStorage primaryCoinStorage;
+    CoinStorage destinationCoinStorage;
+
+
+    @BeforeEach
+    void createCoinStorages() {
+        this.primaryCoinStorage = new CoinStorage(new HashMap<>(), messagePrinter);
+        this.destinationCoinStorage = new CoinStorage(new HashMap<>(), messagePrinter);
+    }
+
+
+    @Test
+    void shouldAddCoin() {
+        Coin coin = Coin.ONEZL;
+
+//        when
+        primaryCoinStorage.addCoin(coin);
+
+//        then
+        assertThat(primaryCoinStorage.getCoins().size()).isEqualTo(1);
+    }
+
+    @Test
+    void shouldFillCoinStorageWithGivenNumberOfCoinsForDenomination() {
+
+//        when
+        primaryCoinStorage.fillCoinStorageWithGivenNumberOfCoinsForDenomination(1);
+
+//        then
+        assertThat(primaryCoinStorage.getCoins().size()).isEqualTo(6);
+        assertThat(primaryCoinStorage.getValueOfCoinsInStorage()).isEqualTo(880);
+    }
+
+    @Test
+    void shouldMoveAllCoinsFromPrimaryToDestinationStorage() {
+
+//        given
+        primaryCoinStorage.fillCoinStorageWithGivenNumberOfCoinsForDenomination(1);
```

Comment:
ok, now I understand why you need to storages - but as this `destinationCoinStorage` is only used in this test - it can be created here

Reply:
Now only one is created there and the other is created separately for purpose of one test